backend/infrahub/core/merge.py

```diff
@@ -105,59 +98,21 @@ async def get_initial_source_branch(self) -> SchemaBranch:
         return self._initial_source_schema
 
     async def has_schema_changes(self) -> bool:
-        graph_diff = await self.get_graph_diff()
-        return await graph_diff.has_schema_changes()
-
-    async def get_schema_diff(self) -> SchemaBranchDiff:
-        """Return a SchemaBranchDiff object with the list of nodes and generics
-        based on the information returned by the Graph Diff.
-
-        The Graph Diff return a list of UUID so we need to convert that back into Kind
-        """
-
-        if self.schema_diff:
-            return self.schema_diff
-
-        graph_diff = await self.get_graph_diff()
-        schema_summary = await graph_diff.get_schema_summary()
-        schema_diff = SchemaBranchDiff()
-
-        # NOTE At this point there is no Generic in the schema but this could change in the future
-        for element in schema_summary.get(self.source_branch.name, []):
-            if element.kind == "SchemaNode" and DiffAction.REMOVED in element.actions:
-                continue
-            node = self.source_schema.get_by_any_id(id=element.node)
-            if isinstance(node, NodeSchema):
-                schema_diff.nodes.append(node.kind)
-            elif isinstance(node, GenericSchema):
-                schema_diff.generics.append(node.kind)
-
-        for element in schema_summary.get(self.destination_branch.name, []):
-            if element.kind == "SchemaNode" and DiffAction.REMOVED in element.actions:
-                continue
-            node = self.destination_schema.get_by_any_id(id=element.node)
-            if isinstance(node, NodeSchema):
-                schema_diff.nodes.append(node.kind)
-            elif isinstance(node, GenericSchema):
-                schema_diff.generics.append(node.kind)
-
-        # Remove duplicates if any
-        schema_diff.nodes = list(set(schema_diff.nodes))
-        schema_diff.generics = list(set(schema_diff.generics))
-
-        self.schema_diff = schema_diff
-        return self.schema_diff
+        diff_summary = await self.diff_repository.summary(
+            base_branch_name=self.destination_branch.name,
+            diff_branch_names=[self.source_branch.name],
+            tracking_id=BranchTrackingId(name=self.source_branch.name),
+            filters={"kind": {"includes": ["SchemaNode", "SchemaAttribute", "SchemaRelationship"]}},
+        )
+        if not diff_summary:
+            return False
+        return bool(diff_summary.num_added or diff_summary.num_removed or diff_summary.num_updated)
```

Comment:
this is the update to `has_schema_changes`
it's already covered by existing tests